project.py

```diff
@@ -114,6 +134,79 @@ def create_trackbars(self):
         cv.createTrackbar('lo', 'result', self.lo, 255, self.trackbar_callback)
         cv.createTrackbar('hi', 'result', self.hi, 255, self.trackbar_callback)
 
+    def check_change(self, image, mask):
+        imageOld = cv.imread("slide"+repr(self.numberOfSlides)+".png",
+                                cv.CV_LOAD_IMAGE_COLOR)
+        maskOld = cv.imread("mask.png", cv.CV_LOAD_IMAGE_GRAYSCALE)
+        if(maskOld is None):
+            return True
+
+        if(mask.shape != maskOld.shape):
+            return False
+
+        applyMask = cv.bitwise_and(maskOld, mask)
+        imageOld = cv.bitwise_and(imageOld, imageOld, mask=applyMask)
+        checkImage = cv.bitwise_and(image, image, mask=applyMask)
+
+        orb = cv.ORB()
+        kp1, ds1 = orb.detectAndCompute(imageOld, None)
+        kp2, ds2 = orb.detectAndCompute(checkImage, None)
+        bf = cv.BFMatcher(cv.NORM_HAMMING, crossCheck=True)
+        matches = bf.match(ds1, ds2)
+
+        val = float(len(matches))/len(kp1)
+        if(val > self.similarity):
+            print "Is the same"
+
+        return False
+
+    def get_sorted_rectangle(self, cnt, image):
+        pts = cnt.reshape(4, 2)
+        rect = np.zeros((4, 2), dtype="float32")
+
+        s = pts.sum(axis=1)
+        rect[0] = pts[np.argmin(s)]
+        rect[2] = pts[np.argmax(s)]
+
+        diff = np.diff(pts, axis=1)
+        rect[1] = pts[np.argmin(diff)]
+        rect[3] = pts[np.argmax(diff)]
+
+        return rect
+
+    def get_croped_image(self, rect, image, mask):
+        (tl, tr, br, bl) = rect
+        widthA = np.sqrt(((br[0] - bl[0]) ** 2) + ((br[1] - bl[1]) ** 2))
+        widthB = np.sqrt(((tr[0] - tl[0]) ** 2) + ((tr[1] - tl[1]) ** 2))
+
+        heightA = np.sqrt(((tr[0] - br[0]) ** 2) + ((tr[1] - br[1]) ** 2))
+        heightB = np.sqrt(((tl[0] - bl[0]) ** 2) + ((tl[1] - bl[1]) ** 2))
+
+        maxWidth = max(int(widthA), int(widthB))
+        maxHeight = max(int(heightA), int(heightB))
+        dst = np.array([
+            [0, 0],
+            [maxWidth - 1, 0],
+            [maxWidth - 1, maxHeight - 1],
+            [0, maxHeight - 1]],
+            dtype="float32")
+
+        warpMat = cv.getPerspectiveTransform(rect, dst)
+        warp = cv.warpPerspective(image, warpMat, (maxWidth, maxHeight))
+        warpMask = cv.warpPerspective(mask, warpMat, (maxWidth, maxHeight))
+        return (warp, warpMask)
+
+    def write_image(self, cnt, image, mask):
+        if(cnt is None):
+            return
+        rect = self.get_sorted_rectangle(cnt, image)
+        warp, warpMask = self.get_croped_image(rect, image, mask)
+
+        if(self.check_change(warp, warpMask)):
+            cv.imwrite("slide" + repr(self.numberOfSlides) + ".png", warp)
```

Comment:
@Foxiy Please try to use [the format specifications](https://docs.python.org/2/library/string.html#formatspec). What you have with `repr` is not nice, nor Pythonic.